SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DynamicSubobjectsTest/DynamicSubobjectsTest.cpp

```diff
@@ -52,198 +64,258 @@ void ADynamicSubobjectsTest::PrepareTest()
-				TArray<AActor*> FoundActors;
-				UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedGASTestActor::StaticClass(), FoundActors);
-				if (FoundActors.Num() == 1)
-				{
-					TestActor = Cast<AReplicatedGASTestActor>(FoundActors[0]);
-					TestActor->TestIntProperty = i + 1;
-				}
-				if (TestActor->TestIntProperty == i + 1)
+				if (AssertIsValid(PlayerCharacter, TEXT("PlayerCharacter should be valid")))
 				{
+					RequireEqual_Vector(PlayerCharacter->GetActorLocation(), CharacterSpawnLocation,
+										TEXT("Client 1 should see themself close to the initial spawn location"), 1.0f);
 					FinishStep();
 				}
-			});
-		}
+			},
+			StepTimeLimit);
+
+		// Step 12 - Client 1 checks it can see the ADynamicSubObjectTestActor
+		AddStep(
+			TEXT("DynamicSubobjectsTestClientCheckIntValueIncreased2"), FWorkerDefinition::Client(1), nullptr, nullptr,
+			[this, i](float DeltaTime) {
+				RequireEqual_Int(TestActor->TestIntProperty, i, TEXT("Client 1 should see the updated TestIntProperty value"));
+				FinishStep();
+			},
+			StepTimeLimit);
 
-		{ // Step 6 - Client 1 checks it can no longer see the AReplicatedGASTestActor
+		if (bLastStepLoop)
+		{
+			// Step 12.1 - Client 1 checks the dynamic component on ReplicatedGASTestActor has been removed
 			AddStep(
-				TEXT("DynamicSubobjectsTestClientCheckIntValueIncreased"), FWorkerDefinition::Client(1), nullptr, nullptr,
-				[this, i](float DeltaTime) {
-					TArray<AActor*> FoundActors;
-					UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedGASTestActor::StaticClass(), FoundActors);
-					if (FoundActors.Num() == 1)
-					{
-						TestActor = Cast<AReplicatedGASTestActor>(FoundActors[0]);
-
-						RequireNotEqual_Int(TestActor->TestIntProperty, i + 1, TEXT("Check TestIntProperty didn't get replicated"));
-						StepTimer += DeltaTime;
-						if (StepTimer >= 0.5f)
-						{
-							FinishStep();
-							StepTimer = 0.0f; // reset for the next time
-						}
-					}
+				TEXT("DynamicSubobjectsTestClientCheckIntValueIncreased2"), FWorkerDefinition::Client(1), nullptr, nullptr,
+				[this](float DeltaTime) {
+					RequireEqual_Int(GetNumComponentsOnTestActor(), InitialNumComponents,
+									 TEXT("ADynamicSubObjectTestActor's dynamic component should have been destroyed."));
+
+					FinishStep();
 				},
 				StepTimeLimit);
 		}
+	}
 
-		{ // Step7 - Server moves Client 1 close to the cube.
-			AddStep(TEXT("DynamicSubobjectsTestServerMoveClient1CloseToCube"), FWorkerDefinition::Server(1), nullptr, [this]() {
-				if (ClientOneSpawnedPawn->SetActorLocation(CharacterSpawnLocation))
-				{
-					if (ClientOneSpawnedPawn->GetActorLocation().Equals(CharacterSpawnLocation, 1.0f))
-					{
-						FinishStep();
-					}
-				}
-			});
-		}
+	// Step 13 - All worker cleanup
+	AddStep(TEXT("DynamicSubobjectsTestAllWorkerCleanup"), FWorkerDefinition::AllWorkers, nullptr, [this]() {
```

Comment:
I would rather this was the first step, ensuring preconditions > postconditions.